src/utils/FileDescriptor.cpp

```diff
@@ -0,0 +1,77 @@
+#include <hyprutils/utils/FileDescriptor.hpp>
+#include <fcntl.h>
+#include <sys/poll.h>
+#include <unistd.h>
+#include <utility>
+
+using namespace Hyprutils::Utils;
+
+CFileDescriptor::CFileDescriptor(int fd) : m_fd(fd) {}
+
+CFileDescriptor::CFileDescriptor(CFileDescriptor&& other) : m_fd(std::exchange(other.m_fd, -1)) {}
+
+CFileDescriptor& CFileDescriptor::operator=(CFileDescriptor&& other) {
+    if (this == &other)
+        return *this;
```

Comment:
this should be an abort(); IMO. the wrapper exists so that fd ownership is clear. Shit will go haywire if there is duplicate ownership

Reply:
changed to abort() , want any error messages of some sorts or similiar?

Reply:
hyprutils cant log